iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -795,12 +763,18 @@ private async Task HandleReceivedCloudToDeviceMessageAsync(MqttApplicationMessag
 
             if (_deviceMessageReceivedListener != null)
             {
-                // We are intentionally not awaiting _deviceMessageReceivedListener callback.
-                // This is a user-supplied callback that isn't required to be awaited by us. We can simply invoke it and continue.
-                _ = _deviceMessageReceivedListener.Invoke(receivedCloudToDeviceMessage);
+                MessageResponse acknowledgementType = await _deviceMessageReceivedListener.Invoke(receivedCloudToDeviceMessage).ConfigureAwait(false);
+
+                // Note that MQTT does not support Abandon or Reject
+                if (acknowledgementType != MessageResponse.Completed)
+                {
+                    if (Logging.IsEnabled)
```

Comment:
I guess we can't throw an exception because the customer won't even observe it, right?

Is there an error processor like what we have for file notifications?

Reply:
Correct.

> I guess we can't throw an exception because the customer won't even observe it, right?

Correct

> Is there an error processor like what we have for file notifications?

We have the connection status callback, but it wouldn't be appropriate to feed it an event like this since it isn't necessarily a connection level problem. My take on cases like this is that users will likely need to re-send the direct method because the service does seem to block on receiving this ack and then block on receiving the response